site/src/modules/apps/apps.ts

```diff
@@ -49,6 +61,73 @@ export const getTerminalHref = ({
 	}/terminal?${params}`;
 };
 
-export const openAppInNewWindow = (name: string, href: string) => {
+export const openAppInNewWindow = (href: string) => {
 	window.open(href, "_blank", "width=900,height=600");
 };
+
+export type GetAppHrefParams = {
+	path: string;
+	host: string;
+	workspace: Workspace;
+	agent: WorkspaceAgent;
+	token?: string;
+};
+
+export const getAppHref = (
+	app: WorkspaceApp,
+	{ path, token, workspace, agent, host }: GetAppHrefParams,
+): string => {
+	if (isExternalApp(app)) {
+		return needsSessionToken(app)
+			? app.url.replaceAll(SESSION_TOKEN_PLACEHOLDER, token ?? "")
+			: app.url;
+	}
```

Comment:
Not a big deal so feel free to ignore but do we need to make all these checks again?  It looks like we already check `isExternalApp(app) && needsSessionToken(app)` in `useAppLink` (which seems to be the only place we use `getAppHref`) and will pass an empty token if the query is disabled. 

IMO it would make more sense to do `getAppHref(...).replace(SESSION_TOKEN_PLACEHOLDER, apiKeyResponse?.key ?? "")` in `useAppLink`.  Although if we are removing this later I suppose it does not matter anyway.

Reply:
Exactly, but I will take a second look into this.